ddl/column.go

```diff
@@ -1526,15 +1531,27 @@ func checkAndApplyNewAutoRandomBits(job *model.Job, t *meta.Meta, tblInfo *model
 // `isDataTruncated` indicates whether the new field and the old field type are the same, in order to be compatible with mysql.
 func checkForNullValue(ctx sessionctx.Context, isDataTruncated bool, schema, table, newCol model.CIStr, oldCols ...*model.ColumnInfo) error {
 	colsStr := ""
+	paramsList := make([]interface{}, 0, 2+len(oldCols))
+	paramsList = append(paramsList, schema.L, table.L)
 	for i, col := range oldCols {
 		if i == 0 {
-			colsStr += "`" + col.Name.L + "` is null"
+			colsStr += "%n is null"
+			paramsList = append(paramsList, col.Name.L)
 		} else {
-			colsStr += " or `" + col.Name.L + "` is null"
+			colsStr += " or %n is null"
+			paramsList = append(paramsList, col.Name.L)
 		}
 	}
-	sql := fmt.Sprintf("select 1 from `%s`.`%s` where %s limit 1;", schema.L, table.L, colsStr)
-	rows, _, err := ctx.(sqlexec.RestrictedSQLExecutor).ExecRestrictedSQL(sql)
+	var buf strings.Builder
+	_, err := fmt.Fprintf(&buf, "select 1 from %%n.%%n where %s limit 1", colsStr)
```

Comment:
Escape twice (one for Fprintf and one for ParseWithParams) reduces readability.
How about:

```
var buf strings.Builder
buf.WriteString("select 1 from %n.%n where ")
for  ... cols {
     buf.WriteString("%n is null") ...
     paramsLIst = append(...)
}
buf.WriteString("limt 1")
```